test/test_multidevice_pipeline.cpp

```diff
@@ -184,10 +186,13 @@ TEST_P(PipelineTestTwoStages, Communication) {
   if (is_stage0_sharded) {
     tv0->axis(0)->parallelize(ParallelType::DIDx);
     tv1->axis(0)->parallelize(ParallelType::DIDx);
+    // Shard outermost-axis of input
+    input_sizes[0] = 1;
```

Comment:
For concrete tensors, the test passes the unsharded size into makeConcreteTensorView. We might want this to be the unsharded size though.

Reply:
Ok I see ! Well spotted :) So it means that with this change we only support concrete tensors and not symbolic shapes.
As far as I am concerned that's not too big of a deal, but I think from a communication pov it can reflect badly to say that we are breaking the support of symbolic size.
Actually the fix shouldn't be too complicated -- on sharded axis, we just need to bind the size of the underlying mesh instead of binding the size of the inputs.